### PR TITLE
Add prow jobs for gardener-extension-registry-cache repo

### PR DESCRIPTION
**Pull request by oliver-goetz**
/kind enhancement

**What this PR does / why we need it**:
This PR adds `prow` jobs for unit tests and image builds of `gardener-extension-registry-cache` repository.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
These are test runs of these jobs:
[ci-gardener-extension-registry-cache-unit](https://prow.gardener.cloud/view/gs/gardener-prow/logs/ci-gardener-extension-registry-cache-unit/1575549905517154304)
[pull-gardener-extension-registry-cache-unit](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener-extension-registry-cache/1/pull-gardener-extension-registry-cache-unit/1575550327208284160)
[pull-gardener-extension-registry-cache-verify-image-build](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener-extension-registry-cache/1/pull-gardener-extension-registry-cache-verify-image-build/1575550679802449920)
[post-gardener-extension-registry-cache-build-images](https://prow.gardener.cloud/view/gs/gardener-prow/logs/post-gardener-extension-registry-cache-build-images/1575551039292051456)

### CODE REVIEW

File: config/jobs/gardener-extension-registry-cache/gardener-extension-registry-cache-build-images.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      serviceAccountName: image-builder
+      containers:
+      - name: image-builder
+        image: eu.gcr.io/gardener-project/ci-infra/image-builder:v20220908-8116cd8
+        command:
+        - /image-builder
+        args:
+        - --log-level=info
+        - --docker-config-secret=gardener-prow-gcr-docker-config
+        - --registry=eu.gcr.io/gardener-project/gardener-extension-registry-cache
```
Comment (timebertt) [inline config/jobs/gardener-extension-registry-cache/gardener-extension-registry-cache-build-images.yaml:23]: Other extensions use the following image repos:
`eu.gcr.io/gardener-project/gardener/extensions/provider-aws` and `eu.gcr.io/gardener-project/gardener/extensions/admission-aws`

We could follow this pattern by specifying `--registry=eu.gcr.io/gardener-project/gardener/extensions/registry-cache` and renaming our Dockerfile targets, correct?

**Review by timebertt — COMMENTED** [review]
One suggestion regarding the image location

File: config/jobs/gardener-extension-registry-cache/gardener-extension-registry-cache-build-images.yaml (hunk 1 above)
Comment (timebertt) [reply config/jobs/gardener-extension-registry-cache/gardener-extension-registry-cache-build-images.yaml]: Done in 35cd955 and https://github.com/gardener/gardener-extension-registry-cache/pull/2

Comment (oliver-goetz) [reply config/jobs/gardener-extension-registry-cache/gardener-extension-registry-cache-build-images.yaml]: `eu.gcr.io/gardener-project/gardener/extensions` looks like an orphan to me.
There are only a couple of images pushed to the subfolders and the youngest one is more than 1 year old

Comment (timebertt) [reply config/jobs/gardener-extension-registry-cache/gardener-extension-registry-cache-build-images.yaml]: What do you mean?
The provider extensions, networking extensions, shoot service extensions all push their images to this repo.
E.g., `eu.gcr.io/gardener-project/gardener/extensions/{provider,admission}-aws:v1.39.0-dev-a75ee7a8a5eaa01567150c8ec51f12379c39d14d` have been pushed very recently.

**Review by timebertt — COMMENTED** [review]
/lgtm

**Review by timebertt — COMMENTED** [review]
/lgtm

**Pull request merged**